File: src/main/java/br/com/blz/testjava/model/repositories/WarehouseRepository.java
package br.com.blz.testjava.model.repositories;

import br.com.blz.testjava.model.entities.Warehouse;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface WarehouseRepository extends JpaRepository<Warehouse, Long> {

    @Query( " select w from Warehouse w"
          + " where w.locality.name = :nameLocality"
          + " and w.product.sku = :sku")
    Optional<Warehouse> findByProductSkuAndNameLocality(@Param("sku")Long sku, @Param("nameLocality") String nameLocality);
}

File: src/main/java/br/com/blz/testjava/domain/repository/WarehouseRepository.kt
package br.com.blz.testjava.domain.repository

import br.com.blz.testjava.domain.entity.Warehouse

interface WarehouseRepository {
  fun save(warehouse: Warehouse, sku: Int): Warehouse

  fun findBySku(sku: Int): List<Warehouse>

  fun deleteBySku(sku: Int)
}
